samples/star/src/commonMain/kotlin/com/slack/circuit/star/petlist/PetListScreen.kt

```diff
@@ -349,18 +350,29 @@ private fun PetListGridItem(
   ) {
     Column(modifier = Modifier.clickable { onClick() }) {
       // Image
-      AsyncImage(
-        modifier = Modifier.fillMaxWidth().testTag(IMAGE_TAG),
-        model =
-          Builder(LocalPlatformContext.current)
-            .data(updatedImageUrl)
-            .memoryCacheKey(animal.imageUrl)
-            .crossfade(AnimationConstants.DefaultDurationMillis)
-            .build(),
-        contentDescription = animal.name,
-        contentScale = ContentScale.Crop,
-        imageLoader = SingletonImageLoader.get(LocalPlatformContext.current),
-      )
+      val imageModifier = Modifier.fillMaxWidth().testTag(IMAGE_TAG)
+      if (animal.imageUrl == null) {
+        Image(
+          rememberVectorPainter(Pets),
+          modifier = imageModifier.padding(8.dp),
+          contentDescription = animal.name,
+          contentScale = ContentScale.Crop,
+          colorFilter = ColorFilter.tint(LocalContentColor.current),
+        )
+      } else {
+        AsyncImage(
```

Comment:
Did the `error`/`fallback` not work on `AsyncImage`?

Reply:
It did ish, but the problem is that we want different padding and tinting on the icon in those cases that I couldn't figure out how to only apply to the error/fallback painter. Do you know a way?

Reply:
Ah true, might be possible by adjusting the modifiers via `onError` or something. Not as simple.